src/core/api/clock.ts

```diff
@@ -292,11 +305,15 @@ function getStalledStatus(
   if (shouldUnstall === true) {
     return null;
   } else if (shouldStall === true || prevStalled !== null) {
-    let reason : "seeking" | "not-ready" | "buffering";
+    let reason : "seeking" | "not-ready" | "buffering" | "internal-seek";
     if (currentState === "seeking" ||
         currentTimings.seeking ||
         prevStalled !== null && prevStalled.reason === "seeking") {
-      reason = "seeking";
+      if (isInternalSeekingComing) {
```

Comment:
I don't think this code works

`currentState` is not equal to `"seeking"` anymore, no?

Reply:
I now take into consideration the previous stalled status.